pkg/cmd/repo/list/list.go

```diff
@@ -58,20 +54,16 @@ func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Comman
 				return cmdutil.FlagErrorf("invalid limit: %v", opts.Limit)
 			}
 
-			if flagPrivate && flagPublic {
-				return cmdutil.FlagErrorf("specify only one of `--public` or `--private`")
-			}
 			if opts.Source && opts.Fork {
 				return cmdutil.FlagErrorf("specify only one of `--source` or `--fork`")
 			}
 			if opts.Archived && opts.NonArchived {
 				return cmdutil.FlagErrorf("specify only one of `--archived` or `--no-archived`")
 			}
 
-			if flagPrivate {
-				opts.Visibility = "private"
-			} else if flagPublic {
-				opts.Visibility = "public"
+			visibility := strings.ToUpper(opts.Visibility)
+			if visibility != "" && (visibility != "PUBLIC" && visibility != "PRIVATE" && visibility != "INTERNAL") {
+				return cmdutil.FlagErrorf("`--visibility` only supports `public`, `private`, or `internal`")
```

Comment:
You could use `cmdutil.StringEnumFlag` when defining the `--visibility` flag so you won't have to perform any explicit checks on the value!